src/runtime/virtcontainers/qemu_amd64.go

```diff
@@ -39,6 +40,12 @@ var qemuPaths = map[string]string{
 	QemuMicrovm: defaultQemuPath,
 }
 
+var kernelRootParams = []Param{
+	{"root", "/dev/pmem0p1"},
+	{"rootflags", "dax,data=ordered,errors=remount-ro rw"},
+	{"rootfstype", "ext4"},
+}
+
```

Comment:
Since both the qemu options and the kernel cmdline ones can be technology specific, the approach that we take in #1590 was to implement those through `enableProtection()` and have a generic `availableGuestProtection()` routine that detects which confidential technology the host is using. Based on `availableGuestProtection()` detection, the specific qemu options and kernel cmdline parameters can be added. See https://github.com/kata-containers/kata-containers/pull/1590/commits/0f5b4d978302c244fefbfc84725bdd5776de8f69#r622022215 for more details, your feedback would be really helpful there.